implementations/CMIP6-UofT/extensions.py

```diff
@@ -0,0 +1,201 @@
+import pystac
+from pystac.extensions.datacube import Dimension, DimensionType, Variable, VariableType
+
+
+class DataCubeHelper:
+    """Return STAC Item from CF JSON metadata, as provided by `xncml.Dataset.to_cf_dict`."""
+
+    axis = {"X": "x", "Y": "y", "Z": "z", "T": "t", "longitude": "x", "latitude": "y", "vertical": "z", "time": "t"}
+
+    def __init__(self, attrs: dict):
+        """
+        Create STAC Item from CF JSON metadata.
+
+        Parameters
+        ----------
+        iid : str
+            Unique item ID.
+        attrs: dict
+            CF JSON metadata returned by `xncml.Dataset.to_cf_dict`.
+        datamodel : pydantic.BaseModel, optional
+            Data model for validating global attributes.
+        """
+        self.attrs = attrs
+
+    def dimensions(self) -> dict:
+        """Return Dimension objects required for Datacube extension."""
+
+        dims = {}
+        for name, length in self.attrs["dimensions"].items():
+            v = self.attrs["variables"].get(name)
+            if v:
+                bbox = self.obj.ncattrs_to_bbox()
+                for key, criteria in coordinate_criteria.items():
+                    for criterion, expected in criteria.items():
+                        if v["attributes"].get(criterion, None) in expected:
+                            axis = self.axis[key]
+                            type_ = DimensionType.SPATIAL if axis in ["x", "y", "z"] else DimensionType.TEMPORAL
+
+                            if v["type"] == "int":
+                                extent = [0, int(length)]
+                            else:  # Not clear the logic is sound
+                                if key == "X":
+                                    extent = bbox[0], bbox[2]
+                                elif key == "Y":
+                                    extent = bbox[1], bbox[3]
+                                else:
+                                    extent = None
+
+                            dims[name] = Dimension(
+                                properties=dict(
+                                    axis=axis,
+                                    type=type_,
+                                    extent=extent,
+                                    description=v.get("description", v.get("long_name", criteria["standard_name"])),
+                                )
+                            )
+
+        return dims
+
+    def variables(self) -> dict:
```

Comment:
Make `@propperty`. Could also use `@functools.cache` to skip recompute.

Reply:
Good tip about `functools.cache`.